packages/react-async/src/Async.tsx

```diff
@@ -134,8 +129,10 @@ export function Async<Value>(props: Omit<Props<Value>, 'manager'>) {
 }
 
 function initialState<Value>(props: Props<Value>): State<Value> {
-  const canResolve = props.defer == null && props.id;
-  const resolved = canResolve && props.id ? tryRequire(props.id()) : null;
+  const resolved = trySynchronousResolve<Value>({
```

Comment:
FWIW, you can usually just pass `props` to this function directly, it has more properties but that's fine because it matches the type signature